docs/DeveloperGuide.adoc

```diff
@@ -137,256 +167,944 @@ image:BetterModelClassDiagram.png[]
+The `**TransactionDate**` is set to the user's current system date in the form "dd/mm/yyyy". The input is now checked in
+a similar fashion as in Step 2 except for:
 
-image::UndoRedoState2.png[]
+* `**Remark**` is parsed by `AddTransactionCommandParser#parseTransactionRemark(ArgumentMultimap)`
+
+Again, since the input is valid, the `**Income**` is successfully added into the transaction list. The transaction list
+now contains 2 `**Transaction**` objects.
+
+image::add-transaction/AddTransaction2.png[]
+
+The following activity diagram summarizes what happens when the user executes a command to add a new `**Transaction**`:
+
+.Activity diagram of adding a transaction into the transaction list
+image::add-transaction/AddTransactionSequence.png[]
+
+==== Design considerations
+
+===== Aspect: Differentiating between `**Expense**` and `**Income**`
+* **Alternative 1: (current choice):** Introduce a `**Transaction**` parent class which both `**Expense**` and `**Income**`
+extends from.
+** Pros: Introduces polymorphism, easing references to either classes by simply referencing the `**Transaction**` object.
+For example, using a single `**List<Transaction>**` instead of needing 2 separate lists `**List<Expense>**` and `**List<Income>**`.
+** Cons: Reduces the readability of the program as polymorphism can be confusing.
+
+* **Alternative 2:** Keep `**Expense**` and `**Income**` classes separate, with each having their own specialized methods.
+** Pros: Maintains an intuitive design: `**Expense**` deducts money and `**Income**` increases money.
+** Cons: Incurs significant overhead and duplicated codes since it is likely that both `**Expense**` and `**Income**` will
+have very similar methods.
+
+===== Aspect: Managing how `**Value**` is stored and handled in `**Expense**` and `**Income**`
+* **Alternative 1: (current choice):** Disallow negative `**Value**` in `**Expense**` object, only using positive amount
+for both `**Expense**` and `**Income**`
+** Pros: Removes the need to implement support for inserting negative `**Value**`. This is due to how `**Value**` constraints
+are applied when restoring `**THRIFT**` data from the data file.
+** Cons: Requires the developer to negate the `**Value**` whenever calculations are done with a `**Expense**` object.
+
+* **Alternative 2:** Allow only negative amount in `**Expense**` object and only positive amount in `**Income**` object
+** Pros: Calculations become trivial: to get balance, sum up the entire `**List<Transaction>**`.
+** Cons: Overhead to parse user input to allow only a single negative symbol and no other symbols.
+
+// tag::taguntag[]
+=== Tag/Untag feature
+==== Implementation
+
+The **tag**/**untag** command performs modifications on existing `**Transaction**` entries. Due to polymorphism, both **Expense**` and `**Income**` objects extends the abstract class `**Transaction**`, and are treated the same way in the context of the **tag**/**untag** command.
```

Comment:
Formatting for Expense and Income objects are different, slightly confusing to read